src/ol/map.js

```diff
@@ -1246,6 +1281,20 @@ ol.Map.prototype.removeOverlay = function(overlay) {
 
 
 /**
+ * This deals with map's overlay collection changes.
+ * @param {ol.Overlay} overlay Overlay.
+ * @protected
+ */
+ol.Map.prototype.removeOverlayInternal = function(overlay) {
```

Comment:
This method is only called once, so you can inline the code in the `ol.CollectionEventType.REMOVE` listener.